test/sync/src/androidTest/kotlin/io/realm/kotlin/test/mongodb/shared/SyncConfigTests.kt

```diff
@@ -498,6 +508,81 @@ class SyncConfigTests {
         assertEquals(10.seconds, config.initialRemoteData!!.timeout)
     }
 
+    @Suppress("invisible_member", "invisible_reference")
+    @Test
+    fun supportedSchemaTypesWhenCreatingSyncConfiguration() {
+        val user = createTestUser()
+        val supportedSchemaTypes = setOf(
+            FlexParentObject::class,
+            FlexChildObject::class,
+            FlexEmbeddedObject::class,
+        )
+
+        val validateConfig = { config: io.realm.kotlin.Configuration ->
+            assertEquals(3, config.schema.size)
+        }
+
+        // Partition-based Sync
+        enumValues<ValueType>().forEach {
+            when (it) {
+                ValueType.STRING -> {
+                    validateConfig(SyncConfiguration.create(user, "", supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, "", supportedSchemaTypes).build())
+                }
+                ValueType.INT -> {
+                    validateConfig(SyncConfiguration.create(user, 1 as Int, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, 1 as Int, supportedSchemaTypes).build())
+                }
+                ValueType.LONG -> {
+                    validateConfig(SyncConfiguration.create(user, 1L, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, 1L, supportedSchemaTypes).build())
+                }
+                ValueType.OBJECT_ID -> {
+                    validateConfig(SyncConfiguration.create(user, ObjectId.create(), supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, ObjectId.create(), supportedSchemaTypes).build())
+                }
+                // ValueType.UUID -> {
+                //     validateConfig(SyncConfiguration.create(user, null as RealmUUID, supportedSchemaTypes))
+                //     validateConfig(SyncConfiguration.Builder(user, null as RealmUUID, supportedSchemaTypes).build())
+                // }
+                ValueType.NULL -> {
+                    validateConfig(SyncConfiguration.create(user, null as String?, supportedSchemaTypes))
+                    validateConfig(SyncConfiguration.Builder(user, null as String?, supportedSchemaTypes).build())
+                }
+                else -> TODO("Test for partition type not defined")
+            }
+
+            // Flexible Sync
+            validateConfig(SyncConfiguration.create(user, supportedSchemaTypes))
+            validateConfig(SyncConfiguration.Builder(user, supportedSchemaTypes).build())
+        }
+
+        // Only do a smoke test for one constructor type as they all go through the same
+        // backing code and testing all combinations didn't seem worth.
+        val config1 = SyncConfiguration.create(user, "", supportedSchemaTypes)
```

Comment:
Any reason for having this as a separate step and not just test it in `validateConfig`?